### PR TITLE
fix(graphql): rds import errors due to primary key and index

### PR DESCRIPTION
**Pull request by sundersc**
- Fixes generate schema issue if a table doesn't contain a primary key.
- Fixes an issue with index directive assigning incorrect datasource for the resolver.
- Fixes use of mysql2 package in RDS lambda.
- Disable populating timestamps for RDS datasource. (We need to populate these fields only if RDS table contains it, it will be addressed in a follow up PR).
- Fixes CREATE, UPDATE, DELETE and GET lambda behavior.

#### Description of how you validated changes
- Manual test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-model-transformer/rds-lambda/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "babel-jest": "^29.1.2",
     "jest": "^29.1.2",
     "knex": "^2.3.0",
-    "mysql2": "^3.1.0",
+    "mysql2": "2.3.3",
```
Comment (marcvberg) [inline packages/amplify-graphql-model-transformer/rds-lambda/package.json:13]: If the new version doesn't support that's alright, any particular reason we can't use the latest mysql2?

**Review by marcvberg — APPROVED** [review]
LGTM, question is not a blocker

File: packages/amplify-graphql-model-transformer/rds-lambda/package.json (hunk 1 above)
Comment (sundersc) [reply packages/amplify-graphql-model-transformer/rds-lambda/package.json:13]: There are some contract changes, I ran into some issues with types with 3.x. Also 2.x is stable and well-tested during POC.

File: packages/amplify-graphql-index-transformer/src/resolvers/resolvers.ts [hunk 2]
```diff
@@ -457,25 +457,35 @@ export function updateResolversForIndex(
...
   if (!(name && queryField)) {
     throw new Error('Expected name and queryField to be defined while generating resolver.');
   }
   const modelName = object.name.value;
   const dbInfo = getDBInfo(ctx, modelName);
-  const dataSourceName = `${object.name.value}Table`;
+  let dataSourceName = `${object.name.value}Table`;
+  if (dbType === MYSQL_DB_TYPE) {
+    dataSourceName = RDSLambdaDataSourceLogicalID;
+  }
```
Comment (alharris-at) [inline packages/amplify-graphql-index-transformer/src/resolvers/resolvers.ts:480]: As a general pattern I don't love setting a default and then mutating it on conditions. I'd prefer this get extracted to a helper so we can avoid the `let`.

File: packages/amplify-graphql-index-transformer/src/resolvers/resolvers.ts [hunk 3]
```diff
@@ -457,25 +457,35 @@ export function updateResolversForIndex(
...
   const dataSource = ctx.api.host.getDataSource(dataSourceName);
   const queryTypeName = ctx.output.getQueryTypeName() as string;
-  const table = getTable(ctx, object);
+
+  let stackId = object.name.value;
+  if (isDynamoDB) {
+    const table = getTable(ctx, object);
+    stackId = table.stack.node.id;
+  }
```
Comment (alharris-at) [inline packages/amplify-graphql-index-transformer/src/resolvers/resolvers.ts:488]: Similar comment here.

**Review by alharris-at — APPROVED** [review]
Style comment, but not blocking.

File: packages/amplify-graphql-index-transformer/src/resolvers/resolvers.ts (hunk 2 above)
Comment (phani-srikar) [reply packages/amplify-graphql-index-transformer/src/resolvers/resolvers.ts:480]: +1 here

**Review by phani-srikar — APPROVED** [review]

**Pull request merged**